mirage/yaml/write_observationlist.py

```diff
@@ -33,117 +218,110 @@ def write_yaml(xml_file, pointing_file, yaml_file, ps_cat_sw=None, ps_cat_lw=Non
-                "      GalaxyCatalog: {}\n".format(GalaxyCatalog),
-                "      ExtendedCatalog: {}\n".format(ExtendedCatalog),
-                "      ExtendedScale: {}\n".format(ExtendedScale),
-                "      ExtendedCenter: {}\n".format(ExtendedCenter),
-                "      MovingTargetList: {}\n".format(MovingTargetList),
-                "      MovingTargetSersic: {}\n".format(MovingTargetSersic),
-                "      MovingTargetExtended: {}\n".format(MovingTargetExtended),
-                "      MovingTargetConvolveExtended: {}\n".format(MovingTargetConvolveExtended),
-                "      MovingTargetToTrack: {}\n".format(MovingTargetToTrack),
-                "      BackgroundRate: {}\n\n".format(BackgroundRate_lw)
+    # assemble string that will constitute the yaml content
+    text_out = ["# Observation list created by write_observationlist.py\n\n"]
+
+    text = ['']
+    entry_number = 0  # running number for every entry in the observation list
+
+
+    observation_numbers = np.unique(xml_dict['ObservationID'])
+    for observation_index, observation_number in enumerate(observation_numbers):
+        first_index = xml_dict['ObservationID'].index(observation_number)
+        text += [
+            "Observation{}:\n".format(observation_number),
+            "  Name: '{}'\n".format(xml_dict['ObservationName'][first_index])
             ]
+        for index in np.where(np.array(xml_dict['ObservationID']) == observation_number)[0]:
+            number_of_dithers = np.int(xml_dict['number_of_dithers'][index])
+            instrument = xml_dict['Instrument'][index]
+            for dither_index in range(number_of_dithers):
+                text += [
+                    "  EntryNumber{}:\n".format(entry_number),
+                    "    Instrument: {}\n".format(instrument),
+                    "    Date: {}\n".format(date),
+                    "    PAV3: {}\n".format(PAV3),
+                    "    DitherIndex: {}\n".format(dither_index),
+                ]
+                if return_dict is None:
+                    return_dict = dictionary_slice(xml_dict, index)
+                else:
+                    return_dict = read_apt_xml.append_dictionary(return_dict, dictionary_slice(xml_dict, index))
+                if instrument in ['NIRCAM', 'WFSC']:
+
+                    # set default values
+                    BackgroundRate_sw = '0.5'
+                    BackgroundRate_lw = '1.2'
+
+                    sw_filt = xml_dict['ShortFilter'][index]
+                    lw_filt = xml_dict['LongFilter'][index]
+
+                    text += [
+                        "    FilterConfig:\n",
+                        # "  FilterConfig{}:\n".format(exposure_index),
+                        "      SW:\n",
+                        "        Filter: {}\n".format(sw_filt),
+                        "        PointSourceCatalog: {}\n".format(
+                            ps_cat_sw[observation_index]),
```

Comment:
if pt_cat_sw is a string rather than a list, then this statement will set the catalog value to the first character of the string.

Reply:
Yes. I need to put in a check like for `catalog_files` in Line 166.